helm: Enable ingress controller in smoke tests (ipv4 + ipv6)

sayboras opened the pull request:
### Description

The original goal is to fix the typo as mentioned in first commit, then to enable ingress controller as part of smoke tests to catch such typo in the future.

sayboras committed:
helm: Correct typo for secret sync in ingress

This commit is to correct wrong attribute in operator role, which causes
missing permission for secret.

Verification was done locally with below installation

```
cilium install \
  --chart-directory ./install/kubernetes/cilium \
  --version v1.12.0-rc1 \
  --kube-proxy-replacement=strict \
  --agent-image "${DOCKER_DEV_ACCOUNT}/cilium:${GIT_HASH}" \
  --operator-image "${DOCKER_DEV_ACCOUNT}/operator-generic:${GIT_HASH}" \
  --helm-set ingressController.enabled=true
```

Relates to ca375a92e13058301a1106489c4e26866b9f8879

Signed-off-by: Tam Mach <tam.mach@cilium.io>

kaworu reviewed (approved):
Good catch @sayboras!

sayboras committed:
gha: Enable ingress controller in smoketest

This commit is to enable ingress controller as part smoke test. While
there is no actual test with Ingress resource right now, this still
helps to catch any issue with helm charts, and make sure cilium pods
(e.g. operator and agent) starts up successfully.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
bpf: Fix complication issue when ENABLE_L7_LB is set

This commit is to fix the below complication issue, when IPv6 only + L7
LB are enabled.

```
level=error msg="Failed to compile bpf_lxc.dbg.o: exit status 1" compiler-pid=3406 linker-pid=3407 subsys=datapath-loader
level=debug msg="/var/lib/cilium/bpf/bpf_lxc.c:201:36: error: member reference type 'struct ipv6_ct_tuple' is not a pointer; did you mean to use '.'?" subsys=datapath-loader
level=debug msg="                cilium_dbg3(ctx, DBG_L7_LB, tuple->daddr.p4, tuple->saddr.p4," subsys=datapath-loader
level=debug msg="                                            ~~~~~^~" subsys=datapath-loader
level=debug msg="                                                 ." subsys=datapath-loader
level=debug msg="/var/lib/cilium/bpf/bpf_lxc.c:201:53: error: member reference type 'struct ipv6_ct_tuple' is not a pointer; did you mean to use '.'?" subsys=datapath-loader
level=debug msg="                cilium_dbg3(ctx, DBG_L7_LB, tuple->daddr.p4, tuple->saddr.p4," subsys=datapath-loader
level=debug msg="                                                             ~~~~~^~" subsys=datapath-loader
level=debug msg="                                                                  ." subsys=datapath-loader
level=debug msg="2 errors generated." subsys=datapath-loader
```

Signed-off-by: Tam Mach <tam.mach@cilium.io>

pchaigno reviewed (approved):
Thanks!
